p2p/conn/connection.go

```diff
@@ -85,6 +85,7 @@ type MConnection struct {
 	config        MConnConfig
 
 	quit       chan struct{}
+	done       chan struct{}
```

Comment:
I'd comment here that closing .quit "causes" .done to close.  Not obvious, but nice convention.